### PR TITLE
TESTS-135: I want to enable calls to a particular branch of GBIT on a PR on any involved repository.

### PR DESCRIPTION
**Pull request by ERafaelMartinez**
Configure gsy-e workflow to call a particular `gsy-backend-integration-tests` branch on a PR.

GBIT repo’s tox file should be modify as well to support this possibility.

**INTEGRATION_TESTS_BRANCH**=feature/GSYE-194

A template to the PR message should be added to help open source users to be aware of this.

### CODE REVIEW

**Review by mclrch — DISMISSED** [review]
LGTM

File: .github/pr_templates/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+TODO: PR description and edit branch pointing to
+      gsy-backend-integration-tests repository.
```
Comment (fievelk) [inline .github/pr_templates/pull_request_template.md:2]: @hannesdiedrich @spyrostz which template do you think we could use? I was thinking of something simple like:
```md
## Reason for the proposed changes

Please describe what we want to achieve and why.

## Proposed changes

-

<!--- If needed, choose which branch of the gsy-backend-integration-tests repository will be used to run integration tests. Please only edit the name of the branch. -->
**INTEGRATION_TESTS_BRANCH**=master
```
What do you think?

Comment (ERafaelMartinez) [reply .github/pr_templates/pull_request_template.md:2]: @spyrostz @hannesdiedrich @fievelk 
I'd like to bring this conversation back to life whenever you have time just to finish this PR.

File: .github/workflows/gsy-e_ci.yaml [hunk 2]
```diff
@@ -9,15 +9,33 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+
+    - uses: 8BitJonny/gh-get-current-pr@1.3.0
```
Comment (hannesdiedrich) [inline .github/workflows/gsy-e_ci.yaml:13]: What a username :-D

File: .github/pr_templates/pull_request_template.md (hunk 1 above)
Comment (hannesdiedrich) [reply .github/pr_templates/pull_request_template.md:2]: I agree with the template Pierpaolo proposes.

**Review by hannesdiedrich — COMMENTED** [review]
Once the template Pierpaolo proposes is added, LGTM

File: .github/pr_templates/pull_request_template.md (hunk 1 above)
Comment (spyrostz) [reply .github/pr_templates/pull_request_template.md:2]: +1 from me too!

**Review by fievelk — DISMISSED** [review]

File: .github/workflows/gsy-e_ci.yaml (hunk 2 above)
Comment (ERafaelMartinez) [reply .github/workflows/gsy-e_ci.yaml:13]: I did not noticed this as I just took Marcelos suggestion and added it on this PR. It is a cool one indeed XD .

File: .github/pr_templates/pull_request_template.md (hunk 1 above)
Comment (ERafaelMartinez) [reply .github/pr_templates/pull_request_template.md]: Awesome, I adapted the template message on [d2bdbe0](https://github.com/gridsingularity/gsy-e/pull/1394/commits/d2bdbe0c4f6f76c4bdbb7a43b523a22d8cb97bb8).

**Comment by ERafaelMartinez** [comment]
Adapted to comments. I appreciate your re-review.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Pull request merged**